### PR TITLE
Lighting

### PR DESCRIPTION
**Pull request by newtvo**
Added skybox shader

### CODE REVIEW

File: ProceduralForest/main.cpp [hunk 1]
```diff
@@ -19,13 +19,161 @@
...
+    glBindBuffer(GL_ARRAY_BUFFER, skyboxVBO);
+    glBufferData(GL_ARRAY_BUFFER, sizeof(skyboxVertices), &skyboxVertices, GL_STATIC_DRAW);
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), (void*)0);
+
+    // load textures
+    // -------------
+    vector<std::string> faces
+            {
+                    FileSystem::getPath("Assets/Textures/skybox/right.jpg"),
```
Comment (mastrHyperion98) [inline ProceduralForest/main.cpp:160]: This throws an error saying the file isnt found

File: ProceduralForest/main.cpp [hunk 2]
```diff
@@ -19,13 +19,161 @@
...
+#include "filesystem.h"
+
+
+using namespace glm;
+using namespace std;
+#define SCREEN_WIDTH 1024.0f
+#define SCREEN_HEIGHT 768.0f
+
+unsigned int loadTexture(const char *path);
+unsigned int loadCubemap(vector<std::string> faces);
```
Comment (mastrHyperion98) [inline ProceduralForest/main.cpp:31]: neither loadCubeMap or loadTextures should be in main. The only function that should be in main is the display loop and the main function.

File: ProceduralForest/Structures/mesh.h [hunk 3]
```diff
@@ -0,0 +1,139 @@
+#ifndef MESH_H
+#define MESH_H
+
+#include <glad/glad.h> // holds all OpenGL type declarations
```
Comment (mastrHyperion98) [inline ProceduralForest/Structures/mesh.h:4]: Project does not use glad

File: ProceduralForest/Structures/model.h [hunk 4]
```diff
@@ -0,0 +1,243 @@
+#ifndef MODEL_H
```
Comment (mastrHyperion98) [inline ProceduralForest/Structures/model.h:1]: File is never used and conflicts with current structure of the project

File: ProceduralForest/Structures/mesh.h [hunk 5]
```diff
@@ -0,0 +1,139 @@
+#ifndef MESH_H
```
Comment (mastrHyperion98) [inline ProceduralForest/Structures/mesh.h:1]: There seems to be a duplicate of this file, although it doesnt seem to be ever imported. It is also a direct copy of what is found at https://learnopengl.com/code_viewer_gh.php?code=includes/learnopengl/mesh.h

without given any attribution to the author or source.

File: ProceduralForest/Structures/filesystem.h [hunk 6]
```diff
@@ -0,0 +1,52 @@
+#ifndef FILESYSTEM_H
```
Comment (mastrHyperion98) [inline ProceduralForest/Structures/filesystem.h:1]: why does this exist? the only place it is used leads to a files not found

File: ProceduralForest/CMakeLists.txt [hunk 7]
```diff
@@ -1,20 +1,31 @@
...
 
+
+
 set(CMAKE_CXX_STANDARD 14)
+set(OpenGlLinkers -lglut -lGLU -lGL) # setting all the Glut libraries as one variable.
 
 find_package(GLEW REQUIRED)
-find_package(glfw3 CONFIG REQUIRED)
-find_package(glm CONFIG REQUIRED)
+find_package(glfw3  REQUIRED)
```
Comment (mastrHyperion98) [inline ProceduralForest/CMakeLists.txt:10]: Was this preventing CMake from finding the libraries? I'm not able to compile the project from your branch it says glew cant be found and using a system shared library leads to shader not compiling

**Review by mastrHyperion98 — CHANGES_REQUESTED** [review]

File: ProceduralForest/Structures/mesh.h (hunk 5 above)
Comment (newtvo) [reply ProceduralForest/Structures/mesh.h:1]: yeah, it was taken directly from it, i havent included the attribution to the author

File: ProceduralForest/Structures/filesystem.h (hunk 6 above)
Comment (newtvo) [reply ProceduralForest/Structures/filesystem.h:1]: it was taken from learnopengl

File: ProceduralForest/CMakeLists.txt (hunk 7 above)
Comment (newtvo) [reply ProceduralForest/CMakeLists.txt:10]: missing the OpenGLLinkers for it

File: ProceduralForest/Structures/model.h (hunk 4 above)
Comment (newtvo) [reply ProceduralForest/Structures/model.h:1]: it supposed to read the cube object as to wrap the skybox around it

File: ProceduralForest/Structures/mesh.h (hunk 3 above)
Comment (newtvo) [reply ProceduralForest/Structures/mesh.h:4]: implemented using LearnOpenGl code

File: ProceduralForest/main.cpp (hunk 2 above)
Comment (newtvo) [reply ProceduralForest/main.cpp:31]: this is for testing only

File: ProceduralForest/main.cpp (hunk 1 above)
Comment (newtvo) [reply ProceduralForest/main.cpp:160]: maybe wrong path!

**Pull request closed without merging**